src/box/lua/connpool.lua

```diff
@@ -94,11 +94,15 @@ local function is_candidate_checked(candidate)
            conn.state == 'closed'
 end
 
-local function connect_to_candidates(candidates)
+-- This method connects to all of the specified instances
+-- and returns the set of successfully connected ones.
+local function acquire_all_candidates(instance_names)
+    if next(instance_names) == nil then return {} end
+
     local delay = WATCHER_DELAY
     local connect_deadline = clock.monotonic() + WATCHER_TIMEOUT
 
-    for _, instance_name in pairs(candidates) do
+    for _, instance_name in pairs(instance_names) do
```

Comment:
Again, what is the reason behind renaming the parameter from `candidates` to `instance_names`?

Reply:
Actually, I've couldn't just leave the naming as it was. 

For the consistency I've changed these names:
- `connect_to_candidates` -> `acquire_all_instances` (vs `acquire_any_candidate`). The parameter is renamed due to the function name no longer talks about candidates at all.
- `get_connection`  -> `select_connection`

Reply:
I usually don't mind some amount of minor readability changes without an explicit separation from the main change. However, here we have more lines of renames/refactorings than lines of the main change.

We have a commit (the first one), which states that it is about the connection liveness check, but I need to read over all these renames to find the 9 lines of code that actually do the job.

Also, we can't see from the commit that the renames are for consistency: the second function with this naming pattern is added in another commit now.

I don't care about a name of this function and a name of its argument (the old and the new variants are meaningful for me both). However, please, separate them from the logic change.

I would even split it more:

* The renames.
* `if next(static_candidates) == nil or next(dynamic_opts) == nil then` -> `if next(instance_names) == nil then return {} end`
* The `skip_connection_check` option and the default behavior change.

Reply:
The `next(dynamic_opts) == nil` check is actually needed to implement the new behavior. So, it is OK to include it into the `skip_connection_check` commit. Initially I interpreted it as a change in the code that doesn't change any user-visible behavior.

Reply:
BTW, `acquire_{all,any}_instance{s,}` doesn't make an impression that it performs connecting. `connect_to_candidates` does.

`connect_all_candidates` and `connect_any_candidate` would be more intuitive.

Or `connect_all_instances` and `connect_any_instance`.

Reply:
I've dropped such unrelated changes after applying Alexander PoC. See here https://github.com/tarantool/tarantool/pull/10438#discussion_r1862483309.

Reply:
Seems resolved, so I'm going to close this discussion.